service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -358,16 +357,16 @@ func (t *transferQueueActiveTaskExecutor) processCancelExecution(
 		return err
 	}
 
-	targetNamespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(primitives.UUIDString(task.GetTargetNamespaceId()))
+	targetNamespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(task.GetTargetNamespaceId())
 	if err != nil {
 		return err
 	}
 	targetNamespace := targetNamespaceEntry.GetInfo().Name
 
 	// handle workflow cancel itself
-	if bytes.Compare(task.GetNamespaceId(), task.GetTargetNamespaceId()) == 0 && task.GetWorkflowId() == task.GetTargetWorkflowId() {
+	if strings.Compare(task.GetNamespaceId(), task.GetTargetNamespaceId()) == 0 && task.GetWorkflowId() == task.GetTargetWorkflowId() {
```

Comment:
here and in other similar changes: a basic `task.GetNamespaceId() == task.GetTargetNamespaceId()` might be more readable?